src/cluster/multi_step_cluster.cpp

```diff
@@ -93,16 +99,34 @@ void MultiStep::find_connected_components(vector<uint32_t>& sindex, unordered_ma
 
 	for (size_t i = 0; i < sindex.size(); i++) {
 		++comp[sindex[i]];
+		sumup_edges[sindex[i]] += nedges[i];
 	}
+
+	return comp;
 }
 
-uint32_t MultiStep::find_max(unordered_map<uint32_t, uint32_t> comp)
-{
-	uint32_t large = 0;
-	for (auto& it : comp) {
-		large = max(large, it.second);
+
+vector<vector<uint32_t>> MultiStep::mapping(unordered_map<uint32_t, uint32_t> sumup_edges) {
```

Comment:
Die Funktion sollte einen aussagekräftigeren Namen haben und als Rückgabe ein Mapping von Zusammenhangskomponente -> Menge berechnen. Der Parameter sollte als const-Referenz übergeben werden.

Reply:
Wäre es hier nicht auch möglich die Zugehörigkeit zu einer Menge als Attribut in das struct mit einzubinden?

Reply:
Ja, das wäre sicher auch möglich.